### PR TITLE
#7258 Fix

### LINKED ISSUE #7258: Admins can spam click the "trigger event" button and cause the game to slow to a crawl.
### Environment
- Game Version:  13-8-2021 develop
- OS Version: Win10 Home.

### Detailed description of bug

Title.

### Steps to reproduce

1. Be admin
2. Spam click the "trigger event" button with random events selected.
3. Profit

### PR DESCRIPTION
**Pull request by Pottingmix**
Added Timer to stop spam on event creation.

Also Added Seconds to GameManager Timer, This would allow Players to properly time bombs and allows for finer tuning of functions.

### Notes:
closes #7258

CL: [Fix] added 5 second cooldown to admin trigger event button

### CODE REVIEW

File: UnityProject/Assets/Scripts/UI/Systems/AdminTools/EventsManagerPage.cs [hunk 1]
```diff
@@ -39,40 +39,58 @@ public void Awake()
...
-		if (!InGameEventType.TryParse(eventTypeDropDown.options[eventTypeDropDown.value].text,
-			out InGameEventType eventType)) return;
+		// Pull time from game manager and put it into a private holder variable which gets cleared on timeout and allows the button to be pressed again. Current WAIT time set to 5 secs
+		
+		stationTimeHolder = GameManager.Instance.stationTime;
 
-		var index = nextDropDown.value;
-
-		if (eventType == InGameEventType.Random)
+		if (stationTimeHolder >= (stationTimeSnapshot))
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/UI/Systems/AdminTools/EventsManagerPage.cs]: Can you unnest to this:

```
if (stationTimeHolder < (stationTimeSnapshot))
   {
            // Tells all admins to wait X seconds, this is based on round time so if the server stutters loading an event it 
            will take it into account effectivly stopping any sort of spam.
			Chat.AddExamineMsgToClient($"Please wait 
             {Mathf.Round((float)stationTimeSnapshot.Subtract(stationTimeHolder).TotalSeconds)} seconds before trying 
              to generate another event.");
            return;
    }
```

Comment (Pottingmix) [reply UnityProject/Assets/Scripts/UI/Systems/AdminTools/EventsManagerPage.cs]: Fixed, as requested.

File: UnityProject/Assets/Scripts/UI/Systems/AdminTools/EventsManagerPage.cs [hunk 2]
```diff
@@ -39,40 +39,60 @@ public void Awake()
...
+			AdminCommandsManager.Instance.CmdTriggerGameEvent(index, isFakeToggle.isOn, announceToggle.isOn, eventType, null);
+		}
 
+		if (stationTimeHolder < (stationTimeSnapshot))
+   		{
+            // Tells all admins to wait X seconds, this is based on round time so if the server stutters loading an event it 
+            // will take it into account effectivly stopping any sort of spam.
+			Chat.AddExamineMsgToClient($"Please wait {Mathf.Round((float)stationTimeSnapshot.Subtract(stationTimeHolder).TotalSeconds)} seconds before trying to generate another event.");
+            return;
+    	}
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/UI/Systems/AdminTools/EventsManagerPage.cs]: Move this to the top of the function

File: UnityProject/Assets/Scripts/UI/Systems/AdminTools/EventsManagerPage.cs (hunk 1 above)
Comment (ThatDan123) [reply UnityProject/Assets/Scripts/UI/Systems/AdminTools/EventsManagerPage.cs]: Then delete this If statement too

**Review by ThatDan123 — APPROVED** [review]

**Review by corp-0 — APPROVED** [review]

**Pull request merged**